torchgeo/datasets/dl4gam_alps.py

```diff
@@ -0,0 +1,450 @@
+
+        # set the local file paths
+        label = f'dataset_{version}'
+        self.fp_archive = self.root / f'{label}.tar.gz'
+        self.dir_patches = self.root / label
+        self.fp_splits_csv = self.root / 'splits.csv'
+
+        # get the corresponding urls and checksums
+        self.url_dataset = self.download_metadata[label]['url']
+        self.md5_dataset = self.download_metadata[label]['checksum']
+        self.url_csv_splits = self.download_metadata['splits_csv']['url']
+        self.md5_csv_splits = self.download_metadata['splits_csv']['checksum']
+
+        self._verify()
+        self._prepare_files()
+
+    def __len__(self) -> int:
+        """The length of the dataset.
+
+        Returns:
+            the number of patches in the dataset
+        """
+        return len(self.fp_patches)
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Load the NetCDF file for the given index and return the sample as a dict.
+
+        Args:
+            index: index of the sample to return
+
+        Returns:
+            dict: a dictionary containing the sample with the following:
+
+                * the Sentinel-2 image (selected bands)
+                * the glacier mask (binary mask with all the glaciers in the current
+                  patch)
+                * the debris mask
+                * the cloud and shadow mask
+                * the additional features (DEM, derived features, etc.) if required
+        """
+        xr = lazy_import('xarray')
+        nc = xr.open_dataset(
+            self.fp_patches[index], decode_coords='all', mask_and_scale=True
+        )
+
+        # extract the S2 image and masks from the netcdf file
+        all_band_names = nc.band_data.long_name
+        idx_img = [all_band_names.index(b) for b in self.bands]
+        image = nc.band_data.isel(band=idx_img).values.astype(np.float32)
+        id_cloud_mask = all_band_names.index('CLOUDLESS_MASK')
+        mask_clouds_and_shadows = ~(nc.band_data.isel(band=id_cloud_mask).values == 1)
+        sample = {
+            'image': image,
+            'mask_glacier': ~np.isnan(nc.mask_all_g_id.values),
+            'mask_debris': (nc.mask_debris.values == 1),
+            'mask_clouds_and_shadows': mask_clouds_and_shadows,
+        }
+
+        # extract the additional features if needed
+        if self.extra_features:
+            for feature in self.extra_features:
+                assert feature in nc, f'Feature {feature} not found in the netcdf file'
+                vals = nc[feature].values.astype(np.float32)
+
+                # impute the missing values with the mean
+                # or zero (for dh/dt and surface velocity)
+                v_fill = 0.0 if feature in ('dhdt', 'v') else np.nanmean(vals)
+                vals[np.isnan(vals)] = v_fill
+
+                sample[feature] = vals
+
+        # convert to torch tensors
+        for k, v in sample.items():
+            sample[k] = torch.from_numpy(v)
```

Comment:
We try to avoid dynamic typing, could we convert to torch tensors earlier or have two different dicts?